src/parse/asp/parser.go

```diff
@@ -32,6 +32,8 @@ type Parser struct {
 	interpreter *interpreter
 	// Stashed set of source code for builtin rules.
 	builtins map[string][]byte
+
+	statements []*Statement
```

Comment:
What is this? I'm a bit unclear why the parser would have a list of statements in it.

Reply:
This is for the bultin statements, as `LoadBuiltins()` already loads the builtins and saves into the Parser.builtin as a map of byte slices, it also loads the builtins as statements. In Analyzer, I use the statements to get the information about the builtin rules. I think since `LoadBuiltins()` already create these statements for the builtin functions and rules, we might as well save these statement for later uses as well.